### ISSUE TITLE
[selectors-4] Consider respecting border-bottom-style on the :visited selector

### ISSUE THREAD

**Issue by sivakusayan** [w3c/csswg-drafts#8871]
I noticed in [this MDN article](https://developer.mozilla.org/en-US/docs/Web/CSS/:visited#privacy_restrictions) that the respected CSS properties on the `:visited` selector are restricted, and cursory testing shows this to be true. I think it would be nice to be able to change the `border-bottom-style` to give another way to differentiate between visited and unvisited links [besides just color](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html). My hunch could be wrong though.

Couldn't find anything [specifically in the specification](https://drafts.csswg.org/selectors/#link), but I'm guessing this was behavior agreed between browser vendors.

**Comment by SelenIT** [comment]
I guess the problem with `border-*-style` for `:visited` styles is that changing it to/from `none` can cause layout changes (since it indirectly changes `border-*-width`), and only properties that never affect layout are considered safe 😳

**Comment by sivakusayan** [comment]
So I just read [this old article](https://dbaron.org/mozilla/visited-privacy) to understand the history of this issue. Really interesting to get a high level overview on it! 

Anyway, in all honesty I'm not sure how important the issue I'm bringing up is, I would rely on the opinions of low vision users for that. Technically speaking, since color is the only allowed differentiator between visited and unvisited links, I think this isn't satisfying the "Use of Color" criteria. Assistive tech seems to have some kind of way of determining what links are visited/unvisited, although I only did cursory testing on Windows. 

An idea - would it make sense to only respect different border styles if neither the visited or unvisited links have `border-style: none`? I think that shouldn't cause layout changes, but I'm not necessarily an expert in CSS.

**Comment by tabatkins** [comment]
Yeah, border-bottom is a layout-effecting property, and thus would trivially expose visited state to the page. Our tricks to hide whether or not a link is visited wouldn't be able to work.

In theory we could change the style between the non-none keywords without causing issues, but using border-style to indicate a state is not something I've ever really seen, so I don't think it's worth trying to innovate here. You can already change the color.

(The correct solution is still to restrict visited-ness to a safe subset that we *can* freely expose, then get rid of all this trickery and nonsense trying (and failing) to hide it. But that's a larger issue.)

**Comment by fantasai** [comment]
You can use `text-decoration-style` to change the underline style. There are a lot of controls over underlines now in https://www.w3.org/TR/css-text-decor-4/

@sivakusayan Does this solve your use case?
